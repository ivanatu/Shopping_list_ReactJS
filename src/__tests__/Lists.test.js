import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ReactDOM from 'react-dom' 
import renderer from 'react-test-renderer'
import List from '../components/lists'
import Addlist from '../components/addlist'
import EditList from '../components/editlist'
import sinon from 'sinon';
import '../setupTests';


describe('<List/>', () => {
    
    const wrapper = mount( <List/> );


    it('has a valid snapshot', () => {
        const component = renderer.create(
        <List/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing', () => {
        shallow(<List/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('calls componentWillMount', () => {
        sinon.spy(List.prototype, 'componentDidMount');
        const wrapper = mount(<List />);
        expect(List.prototype.componentDidMount.calledOnce).toBe(true);
    });
    it('renders the Items class', () => {
        expect(wrapper.find(".List")).toHaveLength(0);
    });
    it('should render <i> without throwing an error', () => {
        expect(wrapper.exists(<i className="fa fa-edit"  />)).toBe(true)
    });
    it('should render <i> without throwing an error', () => {
        expect(wrapper.exists(<i className="fa fa-trash"  />)).toBe(true)
    });
    it('should render <i> without throwing an error', () => {
        expect(wrapper.exists(<i className="fa fa-bars"  />)).toBe(true)
    });
    it('renders three div jsx elements', () => {
        expect(wrapper.find("div")).toHaveLength(10);        
    });
    it('should render a button', () =>{
        expect(wrapper.find('button').length).toEqual(4)
    });
    it('should render <table> without throwing an error', () => {
        expect(wrapper.exists(<table className="table table-hover table-striped"/>))
    });
    it('should render <thead> without throwing an error', () => {
        expect(wrapper.exists(<thead/>))
    });
    it('should render <tr> without throwing an error', () => {
        expect(wrapper.exists(<tr/>))
    });
    it('should render <td> without throwing an error', () => {
        expect(wrapper.exists(<td/>))
    });
    it('should render <tbody> without throwing an error', () => {
        expect(wrapper.exists(<tbody/>))
    });
    it('should render <ul> without throwing an error', () => {
        expect(wrapper.exists(<ul className="pagination justify-content-center"/>))
    });
    
});

describe('<List/> component contains child componenets', () => {
    
        it('should render <Addlist/> component', () => {
            const wrapper = shallow(<Addlist />)
            expect(wrapper.length).toEqual(1)
        });
        it('should render <EditList/> component', () => {
            const wrapper = shallow(<EditList />)
            expect(wrapper.length).toEqual(1)
        });    
});