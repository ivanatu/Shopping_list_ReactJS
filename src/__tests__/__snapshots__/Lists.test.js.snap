// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List/> has a valid snapshot 1`] = `
<div>
  <button
    className="btn btn-success"
    data-target="#myModal"
    data-toggle="modal"
    type="button"
  >
    Add List
  </button>
  <div>
    <div
      className="modal"
      id="myModal"
      role="dialog"
    >
      <div
        className="modal-dialog"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h4>
              Add List
            </h4>
            <button
              className="close"
              data-dismiss="modal"
              id="close2"
              type="button"
            >
              Ã—
            </button>
          </div>
          <div
            className="modal-body"
          >
            <form
              className="register-form"
              method="POST"
              name="login_form"
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                SHOPPING LIST NAME:
                <input
                  className="form-control"
                  id="list"
                  name="list"
                  onChange={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
              </div>
              <button
                className="btn btn-info "
                type="submit"
              >
                Save List
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-inline align-content"
  >
    <form
      className="form-inline align-content"
      onSubmit={[Function]}
    >
      <input
        aria-label="Search"
        className="form-control mr-sm-2"
        name="search"
        placeholder="Search"
        required={true}
        type="search"
      />
      <button
        type="submit"
      >
        <i
          className="fa fa-search"
        />
      </button>
    </form>
  </div>
  <div
    className="container"
  >
    <table
      className="table table-hover table-striped"
    >
      <thead>
        <tr>
          <th
            colSpan="2"
          >
            SHOPPING LIST
          </th>
          <th
            colSpan="10"
          >
            ACTION
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
    <ul
      className="pagination justify-content-center"
    >
      <li
        className="page-item"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <a
          className="page-link"
          id={1}
          onClick={[Function]}
        >
          1
        </a>
      </li>
    </ul>
  </div>
</div>
`;
